src/pipelines.py

```diff
@@ -345,117 +226,79 @@ def mask_postprocessing(model, config, save_output=False):
-                        adapter={'labels': ([('watershed_center', 'detached_images')]),
-                                 },
-                        cache_dirpath=config.env.cache_dirpath,
-                        save_output=save_output)
 
-    binary_fill = Step(name='binary_fill',
-                       transformer=BinaryFillHoles(),
-                       input_steps=[drop_smaller],
-                       adapter={'images': ([('drop_smaller', 'labels')]),
+def mask_postprocessing_borders(model, config, save_output=False):
+    mask_resize = Step(name='mask_resize_borders',
+                       transformer=make_apply_transformer(resize_image,
+                                                          output_name='resized_images',
+                                                          apply_on=['images', 'target_sizes']),
+                       input_data=['input'],
+                       input_steps=[model],
+                       adapter={'images': ([(model.name, 'mask_prediction')]),
+                                'target_sizes': ([('input', 'target_sizes')]),
                                 },
                        cache_dirpath=config.env.cache_dirpath,
                        save_output=save_output)
 
-    return binary_fill
-
-
-def watershed_contours(mask, contour, config, save_output=True):
-    watershed_contour = Step(name='watershed_contour',
-                             transformer=WatershedContour(),
-                             input_steps=[mask, contour],
-                             adapter={'images': ([(mask.name, 'binarized_images')]),
-                                      'contours': ([(contour.name, 'binarized_images')]),
+    category_mapper = Step(name='category_mapper_borders',
+                           transformer=make_apply_transformer(categorize_image,
+                                                              output_name='categorized_images'),
+                           input_steps=[mask_resize],
+                           adapter={'images': ([('mask_resize_borders', 'resized_images')]),
                                       },
-                             cache_dirpath=config.env.cache_dirpath,
-                             save_output=save_output)
-
-    drop_smaller = Step(name='drop_smaller',
-                        transformer=Dropper(**config.dropper),
-                        input_steps=[watershed_contour],
-                        adapter={'labels': ([('watershed_contour', 'detached_images')]),
-                                 },
-
-                        cache_dirpath=config.env.cache_dirpath,
-                        save_output=save_output)
-    return drop_smaller
+                           cache_dirpath=config.env.cache_dirpath,
+                           save_output=save_output)
 
+    borders_filter = Step(name='borders_filter_borders',
+                          transformer=make_apply_transformer(partial(get_channel,
```

Comment:
@taraspiotr those `make_apply_transformer`'s make this code really nice and simple. It was a really good idea @taraspiotr